.github/workflows/e2e.yml

```diff
@@ -0,0 +1,57 @@
+name: e2e Tests
+on:
+  push:
+    branches: [main]
+    paths-ignore:
+      - 'docs/**'  # only trigger build if a file outside of /docs was changed
+      - '_locales/*.xml'
+      - '.vscode/**'
+      - 'README_ASSETS/**'
+      - 'README.md'
+  pull_request:
+    paths-ignore:
+      - 'docs/**'  # only trigger build if a file outside of /docs was changed
+      - '_locales/*.xml'
+      - '.vscode/**'
+      - 'README_ASSETS/**'
+      - 'README.md'
+
+
+
+jobs:
+  test:
+    name: E2E Tests
+    environment: test
+    env:
+      CI: true
+      NODE_ENV: test
+      PRIVATE_CERTIFICATE_KEY: ${{ secrets.PRIVATE_CERTIFICATE_KEY }}
```

Comment:
This are just "dummy" certificates. They have the correct format to be accepted by the server but then are not validated in the Browser (ignoreHTTPSErrors: true).
They are not generated but just read from github secrets.

Reply:
what link2xt suggests would have the advantage of also working in forks, when outside contributors want to contribute. Also it would be easier to follow and understand if it would not depend on GitHub secrets.